File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Chapter 23/CS/WCFServiceCS/App_Code/Service.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.Text;

[ServiceContract()]
public interface IMyService
{
    [OperationContract]
    string MyTask1(string MyValue);
    [OperationContract]
    string MyTask2(DataContract1 dcValue);
}

// NOTE: If you change the class name "Service" here, you must also update the reference to "Service" in Web.config and in the associated .svc file.
public class Service : IMyService
{
    public string MyTask1(string Myvalue)
    {
        return "Hello: " + Myvalue;
    }
    public string MyTask2(DataContract1 dcValue)
    {
        return "Hello: " + dcValue.FirstName;
    }
}
[DataContract]
public class DataContract1
{
    string firstName;
    string lastName;
    [DataMember]
    public string FirstName
    {
        get { return firstName; }
        set { firstName = value; }
    }
    [DataMember]
    public string LastName
    {
        get { return lastName; }
        set { lastName = value; }
    }
}

File: CS/CS.NET-VB.NET/ASP.NET 4.0 Book/Source/Chapter 23/VB/WCFServiceVB/App_Code/Service.vb
Imports System.ServiceModel
Imports System.Runtime.Serialization

' NOTE: If you change the class name "Service" here, you must also update the reference to "Service" in Web.config and in the associated .svc file.

<ServiceContract()> _
Public Interface IMyService
    <OperationContract()> _
    Function MyTask1(ByVal MyValue As String) As String
    <OperationContract()> _
    Function MyTask2(ByVal dcValue As DataContract1) As String
End Interface
Public Class Service
    Implements IMyService
    Public Function MyTask1(ByVal Myvalue As String) As String Implements IMyService.MyTask1
        Return "Hello: " & Myvalue
    End Function
    Public Function MyTask2(ByVal dcValue As DataContract1) As String Implements IMyService.MyTask2
        Return "Hello: " & dcValue.FirstName
    End Function
End Class
<DataContract()> _
Public Class DataContract1
    Private firstName_Renamed As String
    Private lastName_Renamed As String
    <DataMember()> _
    Public Property FirstName() As String
        Get
            Return firstName_Renamed
        End Get
        Set(ByVal value As String)
            firstName_Renamed = value
        End Set
    End Property
    <DataMember()> _
    Public Property LastName() As String
        Get
            Return lastName_Renamed
        End Get
        Set(ByVal value As String)
            lastName_Renamed = value
        End Set
    End Property
End Class
